### PR TITLE
all training parameters taken out

### PR DESCRIPTION
**Pull request by evvahu**
In this PR, all training parameters were taken out of the forward methods. 
The model sets training=False automatically when the model is in eval mode.
This was checked by printing the model's training variable at each point in the training script.
Additionally, I changed the trainig parameter in the classifiers including the transweigh models. In those models, the class variable "self.training" is given to the composition model (as the composition model is always in the same mode (eval or train) as the classifier). I'm not fully sure whether this is necessary but the transweigh method asks for a training parameter and it's probably wrong if we hardcode it.

### CODE REVIEW

**Review by Blubberli — APPROVED** [review]
Thanks you that's perfect!
